core/src/test/java/org/apache/iceberg/hadoop/TestHadoopCatalog.java

```diff
@@ -676,6 +677,27 @@ public void testRegisterExistingTable() throws IOException {
     assertThatThrownBy(() -> catalog.registerTable(identifier, metadataLocation))
         .isInstanceOf(AlreadyExistsException.class)
         .hasMessage("Table already exists: a.t1");
+    assertThatThrownBy(() -> catalog.registerTable(identifier, metadataLocation, false))
+        .isInstanceOf(AlreadyExistsException.class)
+        .hasMessage("Table already exists: a.t1");
+    assertThat(catalog.dropTable(identifier)).isTrue();
+  }
+
+  @Test
+  public void testRegisterAndOverwriteExistingTable() throws IOException {
```

Comment:
This probably shouldn't be supported at all for Hadoop Catalog? Won't this only work if you are replacing the Highest-metadata.json with a Higher-metadata.json?

Reply:
yeah I think this is a good point as Hadoop table operations always look for highest version file in metadata root. 

I feel like maybe we shall not support register table for Hadoop catalog at all, let me override and try throw UnsupportedOperationException  in register-table instead